### PR TITLE
[nrfconnect] Added reading stored onoff state in lighting app

### LINKED ISSUE #22790: [Platform] nrfconnect SVE TC-OO-2.4 fails
### Reproduction steps

This issue is just a copy of problem reported to Telink platform, but in fact applying in the same way to nRFConnect.
See original issue: https://github.com/project-chip/connectedhomeip/issues/22763

### Platform

nrf connect

### Platform Version(s)

_No response_

### Type

Manually tested with SDK

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
#### Issue Being Resolved
* Fixes #22790

#### Change overview
Added for lighting-app reading stored OnOff state on application init.

### CODE REVIEW

**Comment by andy31415** [comment]
> @andy31415 can we cherry-pick this example specific change to v1.0 branch?

Please create cherrypick PRs for me, placing something like `[1.0 Cherrypick]` in the title to be clearly identifiable.

**Review by andy31415 — APPROVED** [review]

File: examples/lighting-app/nrfconnect/main/ZclCallbacks.cpp [hunk 1]
```diff
@@ -70,5 +72,17 @@ void MatterPostAttributeChangeCallback(const chip::app::ConcreteAttributePath &
  */
 void emberAfOnOffClusterInitCallback(EndpointId endpoint)
 {
+    EmberAfStatus status;
+    bool storedValue;
+
+    // Read storedValue on/off value
+    status = Attributes::OnOff::Get(1, &storedValue);
```
Comment (markaj-nordic) [inline examples/lighting-app/nrfconnect/main/ZclCallbacks.cpp:79]: nit: can we avoid the magic number?

**Review by markaj-nordic — APPROVED** [review]

**Review by LuDuda — APPROVED** [review]

File: examples/lighting-app/nrfconnect/main/ZclCallbacks.cpp [hunk 2]
```diff
@@ -70,5 +72,17 @@ void MatterPostAttributeChangeCallback(const chip::app::ConcreteAttributePath &
...
+    if (status == EMBER_ZCL_STATUS_SUCCESS)
+    {
+        // Set actual state to stored before reboot
```
Comment (bzbarsky-apple) [inline examples/lighting-app/nrfconnect/main/ZclCallbacks.cpp:82]: What does this have to do with reboot?

Comment (bzbarsky-apple) [reply examples/lighting-app/nrfconnect/main/ZclCallbacks.cpp:82]: Maybe this is trying to say "set actual state to the state of the cluster that was last persisted" or something?

**Comment by andy31415** [comment]
@kkasperczyk-no - this was merged, you should be able to create the cherrypick PR now.

**Comment by kkasperczyk-no** [comment]
> @kkasperczyk-no - this was merged, you should be able to create the cherrypick PR now.

@andy31415 great, thank you! Here is a link to cherry-pick PR: https://github.com/project-chip/connectedhomeip/pull/22871

**Pull request merged**